engine/table/src/main/java/io/deephaven/engine/table/impl/lang/QueryLanguageParser.java

```diff
@@ -1968,6 +1968,39 @@ static boolean isWideningPrimitiveConversion(Class<?> original, Class<?> target)
         return false;
     }
 
+    public static boolean isLosslessWideningPrimitiveConversion(Class<?> original, Class<?> target) {
```

Comment:
This is interesting. Historically we've allowed widening from Long to Float. I see how this is lossy.

Is this used to disable vectorization? Or is it used to disable lossy widening?

Reply:
See 5.1.2 on https://docs.oracle.com/javase/specs/jls/se9/html/jls-5.html. The intention here is to ensure lossless conversion.